python/pyarrow/tests/test_io.py

```diff
@@ -424,6 +424,12 @@ def ne(a, b):
     eq(buf2, buf5)
 
 
+def test_buffer_eq_bytes():
+    buf = pa.py_buffer(b'some data')
+    assert buf == b'some data'
```

Comment:
Should also check with other bytes-like objects, such as `bytearray(b"some data")`

Reply:
done